tests/test_meta.py

```diff
@@ -254,6 +255,75 @@ def test_commit_files(data_archive, cli_runner):
         assert r.exit_code == 44, r.stderr
 
 
+def test_commit_files_add_and_delete(data_working_copy, cli_runner):
+    with data_working_copy("point-cloud/auckland.tgz") as (repo_dir, wc):
+        # Set the environment variable
+        os.environ["KART_ATTACHMENTS"] = "true"
+        r = cli_runner.invoke(
+            [
+                "commit-files",
+                "-m",
+                "Adding a new file",
+                "my_attachment.txt=My text",
+            ]
+        )
+        assert r.exit_code == 0, r.stderr
+
+        p1 = subprocess.Popen(
+            ["git", "ls-tree", "-t", "-r", "HEAD"], stdout=subprocess.PIPE
+        )
+        p2 = subprocess.Popen(
+            ["grep", "-v", "dataset"], stdin=p1.stdout, stdout=subprocess.PIPE
```

Comment:
This will go away anyway if you make the changes suggested above, but, best practice is to filter python lists using python, eg use list comprehensions.
grep might not be present on every machine, or might work differently on different machines, whereas something like [line for line in all_lines if "dataset" not in line] will always work